iwf/tests/test_internal_channel.py

```diff
@@ -46,6 +48,9 @@ def wait_until(
     ) -> CommandRequest:
         communication.publish_to_internal_channel(test_channel_name3, 123)
         communication.publish_to_internal_channel(test_channel_name4, "str-value")
+        communication.publish_to_internal_channel(
+            test_channel_name_prefix + "abc", "str-value-for-prefix"
+        )
```

Comment:
this confused me a bit, but it looks like we're using this `WaitAnyWithPublishState` state to write to the prefix internal channel and nothing else happens in `WaitAnyWithPublishState` for the prefix internal channel?  But our `WaitAllThenPublishState` will see the write to the `test-internal-channel-prefix-abc` channel and have it's `for_all_command_completed` check satisfied?

Reply:
You are right on the test workflow behavior. 
This test is a bit crazy to test multiple things together. The initState basically goes to two states in parallel: `WaitAnyWithPublishState` and `WaitAllThenPublishState`. 
WaitAllThenPublishState will start with waiting, and publishing on `execute`. WaitAnyWithPublishState will start with publishing+waiting, and then execute. 

that way we can test the publshing on both waitUntil/execute, any/all waitingTypes etc.